### PR TITLE
Recent Pv profiles

### LINKED ISSUE #642: Profile Viewer: Player History
- Could show up to 7 profiles at once
- Up top is the current one
- Uses playerhead as icon, when hovering over the tab display their name
- Could also work with each profile on their own

Concept art: 
image

Idea by J10a1n15
https://canary.discord.com/channels/516977525906341928/693600396492537896/1081336634483675218

### PR DESCRIPTION
**Pull request by CalMWolfs**
Added recent searches into the profile viewer. Closes #642 

image

### CODE REVIEW

**Review by romangraef — APPROVED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/profileviewer/ProfileViewerUtils.java [hunk 1]
```diff
@@ -240,11 +239,12 @@ private static void getPlayerSkull(String username, Consumer<ItemStack> callback
 			callback.accept(null);
 			return;
 		}
+		System.out.println("Searching: " + username);
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/profileviewer/ProfileViewerUtils.java:242]: I don't think we should keep this in.

File: src/main/java/io/github/moulberry/notenoughupdates/profileviewer/ProfileViewerUtils.java [hunk 2]
```diff
@@ -48,12 +48,11 @@
 import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
-import java.util.concurrent.CompletableFuture;
 import java.util.function.Consumer;
 import java.util.stream.Collectors;
 
 public class ProfileViewerUtils {
-	static Map<String, ItemStack> playerSkullCache = new HashMap<>();
+	static final HashMap<String, ItemStack> playerSkullCache = new HashMap<>();
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/profileviewer/ProfileViewerUtils.java:55]: When the concrete implementation is not relevant, one should prefer the interface over the specific type.

**Review by romangraef — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/profileviewer/ProfileViewerUtils.java [hunk 3]
```diff
@@ -52,10 +52,13 @@
...
 
 public class ProfileViewerUtils {
-	static final HashMap<String, ItemStack> playerSkullCache = new HashMap<>();
+	static Map<String, ItemStack> playerSkullCache = new HashMap<>();
 
 	public static JsonArray readInventoryInfo(JsonObject profileInfo, String bagName) {
-		String bytes = Utils.getElementAsString(Utils.getElement(profileInfo, bagName + ".data"), "Hz8IAAAAAAAAAD9iYD9kYD9kAAMAPwI/Gw0AAAA=");
+		String bytes = Utils.getElementAsString(
+			Utils.getElement(profileInfo, bagName + ".data"),
+			"Hz8IAAAAAAAAAD9iYD9kYD9kAAMAPwI/Gw0AAAA="
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/profileviewer/ProfileViewerUtils.java:60]: Btw, this string is still not valid NBT. But at this point i kind of want to keep that in as a meme.

**Review by romangraef — APPROVED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/profileviewer/ProfileViewerUtils.java (hunk 3 above)
Comment (CalMWolfs) [reply src/main/java/io/github/moulberry/notenoughupdates/profileviewer/ProfileViewerUtils.java:60]: yeah I was wondering what it did, but IntelliJ formatted it, I wasn't trying to fix it

**Review by NopoTheGamer — CHANGES_REQUESTED** [review]
I think Soopy nw should also be cached instead of loading it every time you swap

**Comment by romangraef** [comment]
> I think Soopy nw should also be cached instead of loading it every time you swap

I feel like thats more of a job for the ApiCache tho

**Comment by romangraef** [comment]
Plus I'm changing the soopy networth calculation to be in the api proxy soon as well

**Comment by CalMWolfs** [comment]
> I feel like thats more of a job for the ApiCache tho
> Plus I'm changing the soopy networth calculation to be in the api proxy soon as well

Well, I didn't see these but I've made a cache for the networth. It wasn't caching because it was sending data as well

**Comment by romangraef** [comment]
> It wasn't caching because it was sending data as well
Yes, it was intentional that POST requests werent cached, but since soopy is non standard in that way we should have an explicit override for that

**Review by romangraef — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Pull request merged**